zcash_primitives/src/transaction/mod.rs

```diff
@@ -154,14 +170,17 @@ impl TransactionData {
         }
     }
 
+    #[cfg(feature = "zfuture")]
     pub fn zfuture() -> Self {
         TransactionData {
             overwintered: true,
             version: ZFUTURE_TX_VERSION,
             version_group_id: ZFUTURE_VERSION_GROUP_ID,
             vin: vec![],
             vout: vec![],
+            #[cfg(feature = "zfuture")]
             tze_inputs: vec![],
+            #[cfg(feature = "zfuture")]
             tze_outputs: vec![],
```

Comment:
This entire method is under the `zfuture` feature flag, so these are redundant:
```suggestion
            tze_inputs: vec![],
            tze_outputs: vec![],
```